include/seqan3/range/view/translation.hpp

```diff
@@ -33,256 +33,567 @@
+        assert(n < size());
+        switch (data_members->tf)
+        {
+            case translation_frames::FWD_FRAME_0:
+                return translate_triplet((data_members->urange)[n * 3], (data_members->urange)[n * 3 + 1], (data_members->urange)[n * 3 + 2]);
+                break;
+            case translation_frames::REV_FRAME_0:
+                return translate_triplet(complement((data_members->urange)[(data_members->urange).size() - n * 3 - 1]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 2]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 3]));
+                break;
+            case translation_frames::FWD_FRAME_1:
+                return translate_triplet((data_members->urange)[n * 3 + 1], (data_members->urange)[n * 3 + 2], (data_members->urange)[n * 3 + 3]);
+                break;
+            case translation_frames::REV_FRAME_1:
+                return translate_triplet(complement((data_members->urange)[(data_members->urange).size() - n * 3 - 2]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 3]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 4]));
+                break;
+            case translation_frames::FWD_FRAME_2:
+                return translate_triplet((data_members->urange)[n * 3 + 2], (data_members->urange)[n * 3 + 3], (data_members->urange)[n * 3 + 4]);
+                break;
+            case translation_frames::REV_FRAME_2:
+                return translate_triplet(complement((data_members->urange)[(data_members->urange).size() - n * 3 - 3]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 4]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 5]));
+                break;
+            default:
+                throw std::invalid_argument("Invalid type of frame. Choose from FWD_FRAME_0, REV_FRAME_0, FWD_FRAME_1, REV_FRAME_1, FWD_FRAME_2 and REV_FRAME_2.");
+                break;
+        }
+    }
+    //!\}
+
+    //!\brief Implicit conversion to container types.
+    template <typename container_type>
+    operator container_type()
+    //!\cond
+        requires container_concept<std::decay_t<container_type>> &&
+                 std::is_same_v<std::decay_t<value_type>, std::decay_t<ranges::range_value_type_t<container_type>>>
+    //!\endcond
+    {
+        container_type ret;
+        ret.resize(size());
+        std::copy(cbegin(), cend(), ret.begin());
+        return ret;
     }
 };
 
+//!\brief Class template argument deduction for view_translate_single.
+template <typename urng_t>
+//!\cond
+    requires sized_range_concept<urng_t> &&
+             random_access_range_concept<urng_t> &&
+             nucleotide_concept<std::decay_t<ranges::range_reference_t<std::decay_t<urng_t>>>>
+//!\endcond
+view_translate_single(urng_t &&, translation_frames const) -> view_translate_single<urng_t>;
```

Comment:
If you default the first parameter of the constructor, you may also need a second deduction guide.